Json.More/EnumStringConverter.cs

```diff
@@ -94,7 +94,6 @@ public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerial
 					values.Add(immediate);
 					reader.Read();
 				}
-				reader.Read(); // waste the end array
```

Comment:
Actually, this is needed.  It's handling flags enums, which render as arrays.  It mirrors consuming the array start.

The entire array needs to be consumed as part of this.  If there's an error, it's being caused somewhere else.

Reply:
... unless the read on 95 does it?  Probably a good idea to have some tests around this (which I didn't appear to do).

Reply:
Hmm, I'll go check that, though it definitely seemed like an error. I'd be happier if we didn't need to change that, so I'll go verify!

Reply:
Yeah, I'm not sure what led me to that line in the first place anymore; replacing it changed no test results. Thanks for calling it out!

Reply:
I'm seeing an additional 42 tests failing with this added back, many of them "Expected keyword or end of schema object", such as through SelfValidationTests.

Per https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/converters-how-to?pivots=dotnet-7-0#the-use-of-utf8jsonreader-in-the-read-method:
> If you read beyond the end of the object, or if you stop before reaching the corresponding end token, you get a JsonException exception indicating that:
> 
>> The converter 'ConverterName' read too much or not enough.
While I'm not seeing that specific errors, I'm seeing something similar, such as on `RoundTrip("{\"type\":[\"object\",\"string\"]}")`, it reports "Expected token" from the `throw` in `SchemaJsonConverter.Read`.

Reply:
Hm... that would likely be on the `type` keyword (the only place I'm aware that uses flags enums in the serialization).  I'll download the merge branch and poke around a bit.

Reply:
I force-pushed to remove the commit where I removed this line; adding it in did not resolve any tests. Please let me know if there's something else I can look into for this.

I believe existing tests cover this scenario, but were not catching the flaw previously because the converter was being used directly thanks to the extra `JsonSerializer.Deserialize` calls; nothing was using the `Utf8JsonReader` after it had the extra Read.

Reply:
Sounds good.

I've also just found that the converter serializes the zero-value name if there is one 🤦.  I'll deal with that later.